sdk/test-utils/recorder/src/utils.ts

```diff
@@ -498,3 +498,18 @@ export const windowLens: {
     return this.set(propertyPath.slice(1), propertyValue, root[propertyPath[0]]);
   }
 };
+
+/**
+ * Returns true if the given string is a hexadecimal value
+ *
+ * @export
+ * @param {string} value
+ * @returns {boolean}
+ */
+export function isHex(value: string): boolean {
+  // TODO: Add tests to validate regex
+  if (/[0-9A-Fa-f]/g.test(value)) {
```

Comment:
This seems to be accepting more than just hex values.. need to fix.

Reply:
hmm, the question is what if a non-binary response, e.g., plain string, happens to have the same format. Maybe we need to check the content-type too to ensure it's not text/plain? Otherwise I feel that just response doesn't really provide enough information to distinguish the two cases?

Reply:
Yes, I did think of it, wanted to open the discussion.
`transfer-encoding chunked` represents the streams and those are being encoded to "hex" by nock to my understanding, maybe I can distinguish based on that, thoughts?

Reply:
Transfer-Encoding doesn't mean the response is binary See the example at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding.

Reply:
Currently, I'm considering the hex encoding only if the Content-Type is "avro/binary".
It is better to not decode something greedily unintentionally.